src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ContentLengthReadStream.cs

```diff
@@ -174,16 +174,42 @@ private ReadOnlyMemory<byte> ReadFromConnectionBuffer(int maxBytesToRead)
                     return false;
                 }
 
-                while (true)
+                CancellationTokenSource cts = null;
```

Comment:
@geoffkizer, the way this is currently set up, this is going to end up applying to NTLM draining as well, but I noticed there you're explicitly using int.MaxValue as a drain size, so I'm wondering if you'd instead want to be able to pass in an override here for Timeout.Infinite.

Reply:
The NTLM logic is actually using `_pool.Settings._maxResponseDrainSize`, as we do for normal drain.

I think it's fine to use the same settings for normal drain and NTLM.  Originally I had thought we'd need different settings, hence why DrainAsync takes a size parameter.  But we've ended up with fairly generous defaults here, so they seem reasonable to use for NTLM drain as well.

Reply:
Ok.